File: leetcode/solutions/1281.SubtractTheProductAndSumOfDigitsOfAnInteger/subtractProductAndSum.go
package solution

import (
	"math"
)

func subtractProductAndSum(n int) int {
	if n < 10 {
		return 0
	}

	var sum int
	product := 1
	numOfDigits := int(math.Floor(math.Log10(float64(n)))) + 1

	for i := 0; i < numOfDigits; i, n = i+1, n/10 {
		rest := n % 10
		product *= rest
		sum += rest
	}

	return product - sum
}

File: leetcode/solutions/1281.SubtractTheProductAndSumOfDigitsOfAnInteger/subtractProductAndSum.ts
export default function subtractProductAndSum(num: number): number {
  if (num === 0) {
    return 0;
  }

  const numOfDigits = Math.floor(Math.log10(num)) + 1;
  let product = 1;
  let sum = 0;

  for (let i = 0; i < numOfDigits; i++) {
    const rest = num % 10;
    product *= rest;
    sum += rest;
    num = Math.floor(num / 10);
  }

  return product - sum;
}
